src/revivalpmmp/pureentities/entity/monster/walking/Wolf.php

```diff
@@ -401,12 +404,36 @@ public function isFriendly(): bool {
     }
 
     /**
+     * Sets the collar color when tamed
+     *
+     * @param $collarColor
+     */
+    public function setCollarColor($collarColor) {
+        if ($this->isTamed()) {
+            $this->namedtag->CollarColor = new ByteTag(self::NBT_KEY_COLLAR_COLOR, $collarColor); // set collar color
```

Comment:
If you translate code from MCPC, works as SinglePlayer MCPE.

On this subject, entityes, it's same @kniffo80